packages/@aws-cdk/aws-apigatewayv2/lib/http/stage.ts

```diff
@@ -1,30 +1,22 @@
-import { Metric, MetricOptions } from '@aws-cdk/aws-cloudwatch';
-import { Resource, Stack } from '@aws-cdk/core';
+import { Stack } from '@aws-cdk/core';
 import { Construct } from 'constructs';
 import { CfnStage } from '../apigatewayv2.generated';
-import { CommonStageOptions, IDomainName, IStage } from '../common';
+import { CommonStageOptions, StageBase } from '../common';
+import { IApi } from '../common/api';
 import { IHttpApi } from './api';
-import { HttpApiMapping } from './api-mapping';
 
 
 const DEFAULT_STAGE_NAME = '$default';
 
 /**
- * Represents the HttpStage
- */
-export interface IHttpStage extends IStage {
-}
```

Comment:
Would be nice to keep this in case we want to add methods that are applicable to Http stages but not websocket stages. 

Updating a reference from `IStage` to `IHttpStage` is a breaking change, so we can't add this later.
Worth reviewing all your changes from `IHttpStage` -> `IStage` to see if they should be reverted.

Reply:
So if you look closely, the original implementation of `HttpStage` implemented `IStage` instead of `IHttpStage`. I have added back the `IHttpStage` and made the corresponding changes.

Reply:
> the original implementation of HttpStage implemented IStage instead of IHttpStage

Whoops. That was a miss.